inst/examples/06-publishing.Rmd

```diff
@@ -19,42 +19,103 @@ bookdown::publish_book(render = 'local')
+### Before you begin
 
-```bash
-# assume you have initialized the git repository,
-# and are under the directory of the book repository now
+1. You need a GitHub (<https://github.com/>) account.^[If you are not familiar with GitHub, we suggest you read through the free online book "Happy Git with R" (<https://happygitwithr.com/>) first before working through these instructions.]
+
+1. Start with a **bookdown** project linked to a remote GitHub repository that you can push/pull to from your local copy. 
+    
+:::{.rmdimportant latex=true}
+
+If you don't have an existing book, you can create a simple **bookdown** HTML book to practice with instead:
 
-# create a branch named gh-pages and clean up everything
-git checkout --orphan gh-pages
-git rm -rf .
++ Make a new repository on GitHub (<https://happygitwithr.com/new-github-first.html#make-a-repo-on-github-2>),
++ Make a new RStudio Project via git clone: (<https://happygitwithr.com/new-github-first.html#new-rstudio-project-via-git-clone>),
++ From your RStudio Project, run this command in your R console: `bookdown::create_gitbook()` or `bookdown::create_bs4_book()` (depending on the HTML output format you want),
++ Change anything you want (or change nothing) and commit (<https://happygitwithr.com/new-github-first.html#make-local-changes-save-commit-1>),
++ Push your local changes to GitHub (<https://happygitwithr.com/new-github-first.html#push-your-local-changes-to-github>).
 
-# create a hidden file .nojekyll
-touch .nojekyll
-git add .nojekyll
+:::
 
-git commit -m"Initial commit"
-git push origin gh-pages
+
+### Change the output directory of your book
+
+In your local **bookdown** project, open the `_bookdown.yml` file and change the output directory of your book to be `/docs` by adding this line:
+
+```yaml
+output_dir: "docs"
 ```
 
+Save this file and close it.
+
+### Tell GitHub Pages to bypass Jekyll
+
+Create a hidden, empty file named `.nojekyll` from within R in your **bookdown** project root directory:
+
+```
+file.create(".nojekyll")
+```
+
+If you are using RStudio, you may need to refresh the viewer pane to see this new file. You can also use the terminal to create this file with the command `touch .nojekyll`.
+
+This file is [necessary](https://github.blog/2009-12-29-bypassing-jekyll-on-github-pages/) because GitHub Pages supports Jekyll ([jekyllrb.com](http://jekyllrb.com)), a static website builder, by default. You need the `.nojekyll` file because bookdown sites uses files or directories that start with underscores, and these files have a special status within the Jekyll framework. Because the **bookdown** HTML output is a static, standalone website, you need to tell GitHub that your website should *not* be built via Jekyll. 
```

Comment:
This is where the file needs to be inside the `/docs` folder. 

So it should be `file.create("docs/.nojekyll")` in your example. And `docs/` should not be cleaned, otherwise the file would need to be restored.

Other solution would be to create the file at build time, but this currently needs a hack because the file is not moved automatically to output dir. See https://github.com/rstudio/bookdown/issues/1190#issuecomment-871217855
 

This is why we need to provide a better way. 

I would advice your solution for now creating in the right directory. As it will be tracked by Git, a user would quickly see if this will is to be deleted during a cleaning. 

Note on cleaning: You need to clean you book output dir sometimes when working on a book because bookdown won't do that for you. There could be some conflicts sometimes. For example: You work on a chapter, you build, you commit, you are not happy and change the structure of the book which will lead to new html file name because new chapters. The old chapter does not exist anymore, but the HTML files still are. You will commit the new ones but it will not necessarily overwrite the old ones if the names are different. You will have old HTML files in your published book. For this a cleaning is required before rebuilding the book.

Reply:
Got it, I was thinking that was not going to work but then re-read the existing section (https://bookdown.org/yihui/bookdown/github.html) and thought I was wrong.